source/includes/gcp/_firewall_rules.md

```diff
@@ -0,0 +1,198 @@
+
+# The above command returns JSON structured like this:
+```
+```json
+{
+  "data": [
+    {
+      "allowed": [
+        {
+          "IPProtocol": "tcp",
+          "ports": [
+            "80"
+          ]
+        }
+      ],
+      "creationTimestamp": "2019-03-13T13:28:49.840-07:00",
+      "description": "",
+      "direction": "INGRESS",
+      "disabled": false,
+      "id": "4890726785951782638",
+      "kind": "compute#firewall",
+      "logConfig": {
+        "enable": false
+      },
+      "name": "default-allow-http",
+      "network": "https://www.googleapis.com/compute/v1/projects/flo-cmc-gcp-cay/global/networks/default",
+      "priority": 1000,
+      "selfLink": "https://www.googleapis.com/compute/v1/projects/flo-cmc-gcp-cay/global/firewalls/default-allow-http",
+      "sourceRanges": [
+        "0.0.0.0/0"
+      ],
+      "targetTags": [
+        "http-server"
+      ],
+      "action": "allow"
+    }
+  ],
+  "metadata": {
+    "recordCount": 7
+  }
+}
+```
+
+<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/firewallrules</code>
+
+Retrieve a list of all firewall rules in a given [environment](#administration-environments)
+
+ Attributes | &nbsp;
+------- | -----------
+`allowed`<br/>*Array[object]* | The list of ALLOW rules specified by this firewall.
+`creationTimestamp`<br/>*string* | Creation timestamp in RFC3339 text format.
+`denied`<br/>*Array[object]* | The list of DENY rules specified by this firewall.
+`description`<br/>*string* | An optional description of this firewall rule.
+`direction`<br/>*string* | Direction of traffic to which this firewall applies, either INGRESS or EGRESS. The default is INGRESS.
+`disabled`<br/>*boolean* | Denotes whether the firewall rule is disabled. When set to true, the firewall rule is not enforced and the network behaves as if it did not exist.
+`id`<br/>*string* | The unique identifier for the resource.
+`kind`<br/>*string* | Type of the resource.
+`logConfig`<br/>*object* | This field denotes the logging options for a particular firewall rule. If logging is enabled, logs will be exported to Stackdriver.
+`name`<br/>*string* | The display name of the firewall rule
+`network`<br/>*string* | URL of the network resource for this firewall rule.
+`priority`<br/>*integer* | Priority for this rule. This is an integer between 0 and 65535, both inclusive. The default value is 1000. Relative priorities determine which rule takes effect if multiple rules apply. Lower values indicate higher priority. For example, a rule with priority 0 has higher precedence than a rule with priority 1. DENY rules take precedence over ALLOW rules if they have equal priority. Note that VPC networks have implied rules with a priority of 65535. To avoid conflicts with the implied rules, use a priority number less than 65535.
+`selfLink`<br/>*string* | Server-defined URL for this resource.
```

Comment:
Are we systematically providing Google Cloud URLs to other entities? The way this is phrased, I would expect a CloudMC link, not a Google Cloud link.

Reply:
Same comment for `network` on line 70

Reply:
I'll review this

Reply:
The selflink is a variable in every Google object which points at itself. It's like a resource link.

Reply:
I guess if we are consistent in the GCP plugin, this is not a big deal. I just find it weird to see a direct link to the Google Cloud API in CloudMC